### PR TITLE
Feat/a2 469 appeal final comments

### PR DESCRIPTION
**Pull request by DeanH1990**
## Ticket Number

https://pins-ds.atlassian.net/browse/A2-466
https://pins-ds.atlassian.net/browse/A2-467
https://pins-ds.atlassian.net/browse/A2-468
https://pins-ds.atlassian.net/browse/A2-469

## Description of change

gets and displays lpa and appellant final comments  

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		try {
+			return await this.dbClient.appealFinalComment.findMany({
+				where: {
+					caseReference,
+					lpaCode: { not: null }
+				},
+				include: {
+					FinalCommentDocuments: {
+						include: {
+							Document: true
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js:30]: Can we reuse 'DocumentsArgsPublishedOnly' here

File: packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+					caseReference,
+					serviceUserId: { not: null },
+					ServiceUser: {
+						serviceUserType: userType
+					}
+				},
+				include: {
+					FinalCommentDocuments: {
+						include: {
+							Document: true
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js:66]: and here

File: packages/database/src/schema.prisma [hunk 3]
```diff
@@ -606,6 +608,31 @@ model StatementDocument {
   Document        Document        @relation(fields: [documentId], references: [id], onDelete: NoAction, onUpdate: NoAction)
 }
 
+// Stores appeal final comments from either an LPA, appellant or rule 6 group
+model AppealFinalComment {
```
Comment (bryn-pins) [inline packages/database/src/schema.prisma:612]: Can we rename this to just FinalComment
Nice to have the Join table's name be a straight combination of the two tables it's linking between

File: packages/forms-web-app/src/controllers/selected-appeal/final-comments-details/format-final-comment-details.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {AppealFinalComment[]} comments
+ */
+exports.formatFinalComment = (comments) => {
+	return comments.map((comment, index) => {
+		const documents = comment.FinalCommentDocuments?.map((doc) => doc.Document);
+		const documentDetails = formatDocumentDetails(documents, 'appealFinalComment'); //Update when data model set
+
+		const fullText = comment.comments;
+		const truncated = fullText.length > 150;
+		const truncatedText = truncated ? fullText.substring(0, 150) + '...' : fullText;
```
Comment (bryn-pins) [inline packages/forms-web-app/src/controllers/selected-appeal/final-comments-details/format-final-comment-details.js:17]: do I remember this being done somewhere else, either way seems a good candidate for a utility function

**Review by bryn-pins — COMMENTED** [review]
looks good, some minor comments

File: packages/database/src/schema.prisma (hunk 3 above)
Comment (bryn-pins) [reply packages/database/src/schema.prisma:612]: I think they will go in as separate tables?

File: packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js (hunk 1 above)
Comment (DeanH1990) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js:30]: The DocumentsArgsPublishedOnly isnt reusable here as Prisma wont allow a where clause on the indirect Document relation.

Comment (DeanH1990) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js:30]: Created a query which is essentially the same but allows going through the indirect query

Comment (bryn-pins) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js]: couldn't it be reused but just a level lower? Or is that what you tried?
```js
include: {
  FinalCommentDocuments: {
    include: {
	    Document: {
	      where: ...
	      select: ...
	    }
```

Comment (bryn-pins) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js]: I guess that wouldn't filter the join results, this is fine I guess, would just be nice to have the required props/where for display reused

File: packages/database/src/schema.prisma (hunk 3 above)
Comment (DeanH1990) [reply packages/database/src/schema.prisma]: Ah ok, I misunderstood this

> Nice to have the Join table's name be a straight combination of the two tables it's linking between

to mean having the join tables as one, as the table names were already named as comment/statementDocument just needed to rename the comment join table to match

Comment (bryn-pins) [reply packages/database/src/schema.prisma]: sorry yes think I was talking about the fact it was:
AppealFinalComment 
FinalCommentDocument

To change to
FinalComment
FinalCommentDocument

Sorry for misunderstanding and extra work

File: packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js (hunk 1 above)
Comment (DeanH1990) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js]: That was what i tried, from what i found Prisma doesnt like that query as its using a where clause on an indirect relation. What ive put in works on the indirect query

File: packages/database/src/schema.prisma (hunk 3 above)
Comment (DeanH1990) [reply packages/database/src/schema.prisma]: Yea no worries, reading it back now it makes sense. Will fix it

**Review by bryn-pins — DISMISSED** [review]

**Review by bryn-pins — APPROVED** [review]

**Review by bryn-pins — DISMISSED** [review]

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**